### ISSUE TITLE
Service port standard naming for Istio

### ISSUE THREAD

**Issue by mjnagel** [mattermost/mattermost-operator#266]
#### Summary

Currently the service that is created by the operator has two ports, `app` and `metrics` (code [ref](https://github.com/mattermost/mattermost-operator/blob/master/pkg/mattermost/mattermost_v1beta.go#L73)).

Istio (service mesh) utilizes the port names for [automatic protocol detection](https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/) for its sidecars. The important part here is `If the protocol cannot automatically be determined, traffic will be treated as plain TCP traffic`. I ran across this issue with monitoring Mattermost metrics. When you point a prometheus instance at the Mattermost metrics port, it returns a 404 (if Prometheus has Istio sidecars for traffic) due to the "non-standard" port name.

#### Steps to reproduce

This will be a challenging one to reproduce without knowledge and ability to deploy Istio, Prometheus, and Mattermost. I'm happy to walk through a deployment where I'm seeing this issue if that'd be beneficial (video call or similar might be best for this given the amount of detail and complexity).

#### Expected behavior

I would expect that Mattermost metrics can be monitored in any situation, including when my Prometheus instance is Istio injected.

#### Observed behavior (that appears unintentional)

Prometheus returns 404s on the Mattermost endpoints.

#### Possible fixes

I think there's a couple routes that could be taken. For initial testing to validate that this is the issue I worked on [this branch](https://github.com/mjnagel/mattermost-operator/tree/service-port-name), with the changes essentially just renaming the two port names in the service generation (to `http-app` and `http-metrics` so that Istio knows they are HTTP). Confirmed that this works, the only changes required are to the "external" port names, the `targetPort` doesn't need to be changed.

I'd assume this wouldn't be an ideal solution since it could break things for someone potentially. I think ideally the operator provides a way to override the default names of these ports. Example of a simple way to do this:
```yaml
spec:
  service:
    appPortName: "http-app" # defaults to app
    metricsPortName: "http-metrics" # defaults to metrics
```

If we wanted to go further than that and allow more customization on the service (if I'm remember right `serviceAnnotations` are already a value?), we could refactor things into a `service` section in the spec similar to what was done with `Ingress` such as:
```yaml
spec:
  service:
    annotations: ...
    ports:
      - name: "http-app"
         port: 8065
      - name: "http-metrics"
         port: 8067
```

I don't know if that exactly makes sense to do but hopefully that gives an idea of what I'm thinking. Ultimately just looking for a way that makes sense to allow port name overriding and is maintainable/extensible for the future.

**Comment by Szymongib** [comment]
Hey @mjnagel, thanks for reporting the issue.

I have worked with Istio before so I am aware of this behavior.

We are thinking of refactoring CR to define Service customization in a similar way as we did with Ingress (similarly to your second proposition) as we have requests for some other customizations. Not sure when we will be able to go about it as we would probably like to do it in the new CR version so that we can break compatibility for some features like `useServiceLoadbalancer`.

For your case tho, I see that there is another way of defining the protocol, not just by port name but with the use of `appProtocol` on a Service port as mentioned [here](https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/#explicit-protocol-selection). 

I think we could do that just by modifying the Service created by the Operator and holding off with specifying additional customization for now.

Let me know if that would work for you.
